server.py

```diff
@@ -84,167 +126,179 @@ def json_serial(obj):
 
 
 def json_response(content):
-    output = json.dumps(content, default=json_serial)
+    output = json.dumps(object_as_dict(content), default=json_serial)
     return Response(output, mimetype='application/json')
 
 
-def save_response(bound_logger, survey_response):
-    doc = {}
-    doc['survey_response'] = survey_response
-    doc['added_date'] = datetime.utcnow()
+def object_as_dict(obj):
+    return {c.key: getattr(obj, c.key)
+            for c in inspect(obj).mapper.column_attrs}
+
+
+def queue_cs_notification(logger, tx_id):
+    publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_CS_QUEUE)
+    return publisher.publish_message(tx_id)
+
+
+def queue_ctp_notification(logger, tx_id):
+    publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_CTP_QUEUE)
+    return publisher.publish_message(tx_id)
+
+
+def queue_cora_notification(logger, tx_id):
+    publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_CORA_QUEUE)
+    return publisher.publish_message(tx_id)
 
-    invalid_flag = False
 
-    if 'invalid' in survey_response:
-        invalid_flag = survey_response['invalid']
+def save_response(bound_logger, survey_response):
+    invalid = survey_response.get("invalid")
 
     try:
-        result = get_db_responses(invalid_flag).insert_one(doc)
-        bound_logger.info("Response saved", inserted_id=result.inserted_id, invalid=invalid_flag)
-        return str(result.inserted_id), invalid_flag
+        tx_id = survey_response["tx_id"]
+    except KeyError:
+        raise InvalidUsageError("Missing transaction id. Unable to save response",
+                                400)
+    else:
+        response = SurveyResponse(tx_id=tx_id,
+                                  invalid=invalid,
+                                  data=survey_response)
+        try:
+            db.session.add(response)
+            db.session.commit()
+        except SQLAlchemyError:
+            raise server_error("Unable to save response")
+        else:
+            bound_logger.info("Response saved",
+                              inserted_id=tx_id,
+                              invalid=invalid)
+        return invalid
+
+
+def test_sql(connection):
+    """Run a SELECT 1 to test the database connection"""
+    logger.debug("Executing select 1")
+    connection.scalar(select([1]))
 
-    except pymongo.errors.OperationFailure as e:
-        bound_logger.error("Failed to store survey response", exception=str(e))
-        return None, False
+
+@app.errorhandler(500)
+def server_error(error=None):
+    logger.error(error, status=500)
+    message = {
+        'status': 500,
+        'message': error,
+    }
+
+    resp = jsonify(message)
+    resp.status_code = 500
+    return resp
+
+
+@app.errorhandler(InvalidUsageError)
+def invalid_usage_error(error=None):
+    logger.error(error.message, status=400, payload=error.payload)
```

Comment:
No - just the traceback for any error where specified. By it's nature this isn't called when a db succesfully responds.